vault/token_store.go

```diff
@@ -87,6 +91,16 @@ func NewTokenStore(c *Core, config *logical.BackendConfig) (*TokenStore, error)
 	}
 	t.salt = salt
 
+	t.tokenLocks = map[string]*sync.RWMutex{}
+	for i := int64(0); i < 16; i++ {
+		for j := int64(0); j < 16; j++ {
+			t.tokenLocks[fmt.Sprintf("%s%s",
+				strconv.FormatInt(i, 16),
+				strconv.FormatInt(j, 16))] = &sync.RWMutex{}
+		}
+	}
+	t.tokenLocks["global"] = &sync.RWMutex{}
```

Comment:
Discussed in my feedback elsewhere; this is for tokens using custom IDs. I will rename to "custom" to be more accurate.